### DISCUSSION TITLE
Can I use libgit2 in commercial purposes?

### DISCUSSION THREAD

**Discussion by alex-aparin** [libgit2/libgit2#5876] [Q&A]
I want to embed libgit2 in my commercial application. I want to create static library of libgit2 and link it with my application. Is there any restrictions for my application? something like should I provide source code of my program by user request?

**Comment by ethomson** [comment]
I am not a lawyer and cannot provide you legal advice.  The libgit2 team would always encourage you to consult a lawyer for the specifics of your application and requirements.

Having said that, our license is the GPLv2 with a linking exception.  The goal of this license is that libgit2 itself is GPLv2 but you can use it within your application, even if your application is _not_ GPLv2.  Meaning that by distributing libgit2, you _do_ need to make the source for libgit2 available upon request.  But you may link libgit2 into your application that is not GPLv2, and there's no obligation for you to do anything with the source to your independent application.

**Comment by alex-aparin** [comment]
@ethomson , thanks!
